ferveo/src/bindings_wasm.rs

```diff
@@ -99,69 +99,49 @@ fn unwrap_messages_js(
     Ok(messages)
 }
 
+macro_rules! generate_common_methods {
+    ($struct_name:ident) => {
+        #[wasm_bindgen]
+        impl $struct_name {
+            #[wasm_bindgen(js_name = "fromBytes")]
+            pub fn from_bytes(bytes: &[u8]) -> JsResult<$struct_name> {
+                from_js_bytes(bytes).map(Self)
+            }
+
+            #[wasm_bindgen(js_name = "toBytes")]
+            pub fn to_bytes(&self) -> JsResult<Vec<u8>> {
+                to_js_bytes(&self.0)
+            }
+
+            #[wasm_bindgen]
+            pub fn equals(&self, other: &$struct_name) -> bool {
+                self.0 == other.0
```

Comment:
If the underlying struct doesn't derive `PartialEq/Eq`, the macro would throw at compile time, i.e. macros expanded and then type-checked just like any other Rust code.

But on the other hand, I'm assuming that each of these is "a tuple of one element", so perhaps I should test for `self == other`

Reply:
> the macro would throw at compile time, i.e. macros expanded and then type-checked just like any other Rust code.

So it currently works with the objects there, cool. However, thinking long-term and for code maintenance, I think that the `equals` macro should be separate to at least provide better optionality.

Interested in @fjarri 's thoughts here...maybe I'm just being overly cautious.

Reply:
Ok, I see what you mean. If it comes to this then maybe we could revisit this macro and refactor it into something that permits optionality.

As a side note, I'm starting to develop an opinion that all structures exposed by bindings should implement all methods you would expect from their counterpart in the given language. With an exception for cases where there is some higher power (security) that demands a different interface.

Reply:
A separate equality derivation would be cleaner, but if all the existing objects need both serialization and equality, the merged variant would work. It's not public interface anyway, so it can be updated at any point.